packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -203,19 +234,21 @@ class NetworkController {
       _recordingStateInitializedForIsolates = true;
       await _networkService.initializeRecordingState();
     }
-    updatePollingState(_httpRecordingNotifier.value);
+    updatePollingState(_recordingNotifier.value);
   }
 
   void removeClient() {
     _countActiveClients--;
-    updatePollingState(_httpRecordingNotifier.value);
+    updatePollingState(_recordingNotifier.value);
   }
 
-  /// Clears the previously collected HTTP timeline events and resets the last
-  /// refresh timestamp to the current time.
+  /// Clears the previously collected HTTP timeline events, clears the socket
+  /// profile from the vm, and resets the last refresh timestamp to the current
+  /// time.
   Future<void> clear() async {
     await _networkService.updateLastRefreshTime();
-    _requests.value = HttpRequests();
-    _selectedHttpRequest.value = null;
+    await networkService.clearSocketProfile();
```

Comment:
can we have one networkService method that does all the clearing rather than having updateLastRefreshTime and clearSockerProfile?

Reply:
done